homeassistant/components/rointe/climate.py

```diff
@@ -0,0 +1,261 @@
+        if (
+            self._radiator.mode == RADIATOR_MODE_AUTO
+            and self._radiator.preset == HVACMode.OFF
+        ):
+            return HVACAction.IDLE
+
+        # Forced to off, either on Manual or Auto mode.
+        if not self._radiator.power:
+            return HVACAction.OFF
+
+        # Otherwise, it's heating.
+        return HVACAction.HEATING
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Convert the device's preset to HA preset modes."""
+
+        if self._radiator.preset == RADIATOR_PRESET_ECO:
+            return PRESET_ECO
+        if self._radiator.preset == RADIATOR_PRESET_COMFORT:
+            return PRESET_COMFORT
+        if self._radiator.preset == RADIATOR_PRESET_ICE:
+            return PRESET_ROINTE_ICE
+
+        # Also captures "none" (man mode, temperature outside presets)
+        return None
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temperature = float(kwargs["temperature"])
+
+        if not await self.device_manager.send_command(
+            self._radiator, CMD_SET_TEMP, target_temperature
+        ):
+            LOGGER.error(
+                "Failed to set Temperature [%s] for [%s]",
+                target_temperature,
+                self._radiator.name,
+            )
+
+            raise HomeAssistantError(
+                f"Failed to set HVAC mode for {self._radiator.name}"
+            )
+
+        await self._signal_thermostat_update()
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+
+        LOGGER.debug("Setting HVAC mode to %s", hvac_mode)
+
+        if not await self.device_manager.send_command(
+            self._radiator, CMD_SET_HVAC_MODE, hvac_mode
+        ):
+            LOGGER.error(
+                "Failed to set HVAC mode [%s] for [%s]", hvac_mode, self._radiator.name
+            )
+
+            raise HomeAssistantError(
+                f"Failed to set HVAC mode for {self._radiator.name}"
+            )
+
+        await self._signal_thermostat_update()
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new target preset mode."""
+        LOGGER.debug("Setting preset mode: %s", preset_mode)
+
+        if not await self.device_manager.send_command(
+            self._radiator, CMD_SET_PRESET, preset_mode
+        ):
+            LOGGER.error(
+                "Failed to set preset mode [%s] for [%s]",
+                preset_mode,
+                self._radiator.name,
+            )
+
+            raise HomeAssistantError(
+                f"Failed to set HVAC mode for {self._radiator.name}"
+            )
+
+        await self._signal_thermostat_update()
+
+    async def _signal_thermostat_update(self):
+        """Signal a radiator change."""
+
+        # Update the data
+        await self.coordinator.async_request_refresh()
```

Comment:
`async_request_refresh` uses the coordinator's debouncer, which will wait about 10 seconds to before updating (to see if multiple calls can be batched). Is that what you want? (totally fine if so—just checking.)

Reply:
I wasn't aware of that delay. Honestly I'd prefer that the refresh request is dispatched immediately. (If possible)

The objective here is to request the device's status (from the API) after sending a request to modify something.

The process of modifying a parameter in this integration is:

**Note:** _On the text below "device" means the physical hardware being controlled. not the HA object._

1. The user changes something on the UI (example `climate.async_set_temperature()`)
2. Send the request to the API via the device_manager.
3. Assume the request will be successful (optimistic approach).
4. Update the underlying data object that backs the entities with the new value (for example set the new temperature and other props like hvac_mode)
  i. This will update the UI with the new value. (more on this at the end)
5. Submit the request refresh. (`coordinator.async_request_refresh()`) to load the current state of the device.
   1. In case anything goes wrong on the API the entity's state will be updated to reflect the true state of the device.

Essentially this is to defend against a scenario where a successful call to API (HTTP 200) is rejected by the device for some reason and does not change its state.

If you prefer a different approach I'll be happy to change the code.

Additional question: How could I force the entities to refresh? For example, on my climate platform I have:
```

    @property
    def current_temperature(self) -> float:
        """Get current temperature (Probe)."""
        return self._radiator.temp_probe
```

When the user sets a new temperature I need set other properties like hvac_mode. I notice that it takes a few seconds for those changes to be reflected on the UI. How could I ensure the UI is updated, considering I updated the underlying object that backs all of the platform's properties ?

Reply:
> I wasn't aware of that delay. Honestly I'd prefer that the refresh request is dispatched immediately. (If possible)

In that case, you can use `async_refresh`.

> Additional question: How could I force the entities to refresh?

The common way to do this when using a `DataUpdateCoordinator` is to have all of your entities inherit from (`CoordinatorEntity`)[https://github.com/home-assistant/core/blob/d423cbf8ebc3fec1303055b79e7006cc0701c94a/homeassistant/helpers/update_coordinator.py#L349]. By default, these entities subscribe to coordinator updates and update whenever new data is received: https://github.com/home-assistant/core/blob/d423cbf8ebc3fec1303055b79e7006cc0701c94a/homeassistant/helpers/update_coordinator.py#L369-L376

Reply:
Thanks! I've changed to `async_refresh`. My entities already inherit from `CoordinatorEntity`.